guardrails/embedding.py

```diff
@@ -129,3 +136,60 @@ def output_dim(self) -> int:
             return 12288
         else:
             raise ValueError("Unknown model")
+
+
+class ManifestEmbedding(EmbeddingBase):
+    def __init__(
+        self,
+        client_name: str = "openai",
+        client_connection: Optional[str] = None,
+        cache_name: Optional[str] = None,
+        cache_connection: Optional[str] = None,
+        engine: Optional[str] = "text-embedding-ada-002",
+        encoding_name: Optional[str] = "cl100k_base",
+        max_tokens: Optional[int] = 8191,
+    ):
+        try:
+            from manifest import Manifest
+        except ImportError:
+            raise ImportError(
+                "The `manifest` package is not installed. "
+                "Install with `pip install manifest-ml`"
+            )
+        super().__init__(engine, encoding_name, max_tokens)
+        self._client_name = client_name
+        self._client_connection = client_connection
+        self._cache_name = cache_name
+        self._cache_connection = cache_connection
+
+        manifest_args = {
+            "client_name": client_name,
+            "client_connection": client_connection,
+            "cache_name": cache_name,
+            "cache_connection": cache_connection,
+            "engine": engine,
+        }
+        manifest_args = {k: v for k, v in manifest_args.items() if v is not None}
+        self._manifest = Manifest(**manifest_args)
+
+    def embed(self, texts: List[str]) -> List[List[float]]:
+        embeddings = []
+        for text in texts:
+            embeddings.append(
+                super()._len_safe_get_embedding(text, self._get_embedding)
+            )
+
+        return embeddings
+
+    def embed_query(self, query: str) -> List[float]:
+        resp = self._get_embedding([query])
+        return resp[0]
+
+    def _get_embedding(self, texts: List[str]) -> List[float]:
+        embeddings = self._manifest.run(texts)
+        return embeddings.tolist()
+
+    @property
```

Comment:
It would be great to cache this! https://docs.python.org/3/library/functools.html#functools.cached_property